udata_piwik/metrics.py

```diff
@@ -1,123 +1,170 @@
-    model = Organization
-    name = 'dataset_views'
-    display_name = _('Datasets views')
-
-    def get_value(self):
-        ids = [d.id for d in
-               (Dataset.objects(organization=self.target).only('id') or [])]
-        return int(Metrics.objects(object_id__in=ids, level='daily')
-                          .sum('values.nb_uniq_visitors'))
-
-
-class OrgResourcesDownloads(Metric):
-    model = Organization
-    name = 'resource_downloads'
-    display_name = _('Resources downloads')
-
-    def get_value(self):
-        ids = itertools.chain(*[
-            [getattr(r, 'id', None) for r in d.resources or []] for d in
-            (Dataset.objects(organization=self.target).only('resources') or [])
-        ])
-        return int(Metrics.objects(object_id__in=ids, level='daily')
-                          .sum('values.nb_uniq_visitors'))
-
-
-class OrgReusesViews(Metric):
-    model = Organization
-    name = 'reuse_views'
-    display_name = _('Reuses views')
-
-    def get_value(self):
-        ids = [d.id for d in
-               (Reuse.objects(organization=self.target).only('id') or [])]
-        return int(Metrics.objects(object_id__in=ids, level='daily')
-                          .sum('values.nb_uniq_visitors'))
-
-
-def aggregate_datasets_daily(org, day):
-    keys = ['datasets_{0}'.format(k) for k in KEYS]
-    ids = [d.id for d in Dataset.objects(organization=org).only('id')]
-    metrics = Metrics.objects(object_id__in=ids,
-                              level='daily', date=day.isoformat())
-    values = [int(metrics.sum('values.{0}'.format(k))) for k in KEYS]
-    return Metrics.objects.update_daily(org, day, **dict(zip(keys, values)))
-
-
-def aggregate_reuses_daily(org, day):
-    keys = ['reuses_{0}'.format(k) for k in KEYS]
-    ids = [r.id for r in Reuse.objects(organization=org).only('id')]
-    metrics = Metrics.objects(object_id__in=ids,
-                              level='daily', date=day.isoformat())
-    values = [int(metrics.sum('values.{0}'.format(k))) for k in KEYS]
-    Metrics.objects.update_daily(org, day, **dict(zip(keys, values)))
-
-
-def upsert_metric_for_day(obj, day, data):
-    oid = obj.id if hasattr(obj, 'id') else obj
-    if not isinstance(day, str):
-        day = (day or date.today()).isoformat()
-    commands = dict(('inc__values__{0}'.format(k), data[k]) for k in KEYS)
-    metrics = Metrics.objects(object_id=oid, level='daily', date=day)
-    return metrics.update_one(upsert=True, **commands)
-
-
-def clear_metrics_for_day(day):
-    if not isinstance(day, str):
-        day = (day or date.today()).isoformat()
-    commands = dict(('unset__values__{0}'.format(k), 1) for k in KEYS)
-    metrics = Metrics.objects(level='daily', date=day)
-    return metrics.update(upsert=False, **commands)
+def update_metrics_from_backend():
+    '''
+    TODO: factorize those fns (maybe class based?)
+    '''
+    update_resources_metrics_from_backend()
+    update_datasets_metrics_from_backend()
```

Comment:
@quaxsze we need to add the functions for the missing objects: User (maybe), Reuse, Organization.

And maybe handle the TODO if you have an idea. The code from `update_resources_metrics_from_backend` and `update_datasets_metrics_from_backend` below is very similar.